python/neuroglancer/ingest/volumes.py

```diff
@@ -55,85 +65,66 @@ def underlying(self):
 
     @property
     def num_channels(self):
-        if len(self.shape) == 3:
+        if len(self._shape) == 3:
             return 1
-        elif len(self.shape) == 4:
-            return self.shape[0]
+        elif len(self._shape) == 4:
+            return self._shape[-1]
         else:
-            raise Exception('Wrong shape')  
+            raise Exception('Wrong shape')
+
+    @property
+    def encoding(self):
+        return 'raw'
+      
     
 class HDF5Volume(Volume):
 
-    def __init__(self, path):
-        import h5py
+    def __init__(self, path, layer_type):
         self._f = h5py.File(path, 'r')
         self._data = self._f['main']
-        self._layer_type = 'image'
-        self._mesh = False
-        self._resolution = [4,4,40]
-        self._underlying = self.shape
-        self._data_type = self._f['main'].dtype
 
-    @property
-    def shape(self):
-        return self._data.shape[::-1]
+        self._layer_type = layer_type
+        
+        if self._layer_type == 'image':
```

Comment:
could save 5 lines by writing this as: `self._mesh = (self._layer_type == 'segmentation')`